substrate/utils/prometheus/src/lib.rs

```diff
@@ -111,7 +111,11 @@ async fn init_prometheus_with_listener(
 		}
 	});
 
-	let server = Server::builder(listener).serve(service);
+	// Shutdown server when "signal" is dropped
+	let (_signal, on_exit) = exit_future::signal();
```

Comment:
```suggestion
	let (signal, on_exit) = tokio::sync::oneshot::channel::<()>();
```

Reply:
I also prefer the one shot chan solution